R/modules.R

```diff
@@ -194,13 +228,15 @@ module <- function(label = "module",
                    filters,
                    datanames = "all",
                    server_args = NULL,
-                   ui_args = NULL) {
+                   ui_args = NULL,
+                   type = c("teal_module", "teal_module_reporter", "teal_module_landing")) {
```

Comment:
```suggestion
                   type = c("teal_module", "teal_module_landing", "teal_module_reporter")) {
```
Moving reporter to last because it's used internally (I think).

Reply:
@chlebowa what do you mean by "used internally"? `reporter_previewer_module` is an exported module in teal.
@m7pr  please remember to change a constructor call in `reporter_previewer_module` to use a `type` argument instead of [this](https://github.com/insightsengineering/teal/blob/a71f06cb04b6101d9421643b92567ffde5d964cc/R/reporter_previewer_module.R#L35)

Existence of `reporter_previewer_module` provoke me to think where these "special" modules should be placed. In `teal` or `teal.modules.general`?

Reply:
I mean an app dev does not create one by hand but one is created by `teal` if any of the modules has a reporter.

Reply:
Yes `reporter_previewer_module` is added automatically if any `teal_module` has reporter argument. But also,  
`reporter_previewer_module` can be included by app developer in the list of modules. App developer have ability to change previewer within exposed arguments `label` and `server_args`.

Reply:
In that case feel free to ignore the suggestion. But it wouldn't hurt to arrange the classes in the order of descending use frequency.

Reply:
I cant really tell whats more frequent, reporter or landing popup. landing is new, reporter has been seen in most exposed apps on shinyapps. That really doesnt matter.

Reply:
@m7pr @chlebowa how do you feel, where `reporter_previewer_module` should be placed? In `teal` or in `teal.modules.general`? I think both should be in `teal`. `module(type = )` argument is a fiction, because at this moment nobody is able to make own previewer or landing page module. We have no instructions how to make special modules). Exposing `type` just because `tm_landing_popup` is in tmg isn't good solution in my opinion. This is why I think tm_landing_page should be in `teal`. Any thoughts?

Reply:
I had a similar sentiment before but @m7pr and @lcd2yyz thought it was better placed in tmg with the other modules.

Reply:
I actually dont matter, and from the implementation perspective this is OK to have it in teal, because we have a previewer in here. I think it's fine to have it in teal. What are your thoughts @lcd2yyz and @pawelru

Reply:
>  @m7pr and @lcd2yyz thought it was better placed in tmg with the other modules.

Ok, but it is not for free, it complicates `module` definition because of `type` argument. I suggest to move `tm_landing_page` to `teal` and remove `type` argument from `module` function.

Reply:
I don't mind having the two special modules being created with
```
special_module_template_function <- function(...) {
  ...
  ans <- module(...)
  class(ans) <- c("special_class", class(ans))
  ans
}
```

[@pawelru disagrees](https://github.com/insightsengineering/teal/pull/934#issuecomment-1766384682)

Reply:
so we do assign a class for `tm_landing_page` the same way we assign class for `reporter_previewer_module` ?

Reply:
> @pawelru [disagrees](https://github.com/insightsengineering/teal/pull/934#issuecomment-1766384682)

Yes but I believe part of this opinion was driven by hidden reversed dependency between setting class in tmg and internals of teal. Here everything is in `teal` internals and relevant comment should give us developers sufficient hint what is going on.

Reply:
> @pawelru [disagrees](https://github.com/insightsengineering/teal/pull/934#issuecomment-1766384682)

Yes but I believe part of this opinion was driven by hidden reversed dependency between setting class in tmg and internals of teal. In suggestion everything stays in `teal` (internals) and relevant comment should give us developers sufficient hint what is going on.

Reply:
just moved `tm_landing_popup` to `teal` and removed `type` argument from `module`